### PR TITLE
Add metrics filtering to devices query

### PR DESCRIPTION
**Pull request by elinol**
Implements filtering on metric values.

What we want is to:
get all devices where the latest associated device metric for specified key is matching the provided filter (greater or less than X)

<s>The current query gives the latest device metric where the requirements are met, so if there are at least one metric report that match for a device, that device will be included in result, no matter the timestamp.</s>

Queries the latest inserted record for given metrics key, and checks if it matches the filter.

Includes a simple little functionality for testing in browser, just add params like `?metric=cpu_temp&metric_operator=gt&metric_value=37.0` to devices URL.

### CODE REVIEW

**Comment by joshk** [comment]
@elinol did you want a code review on this PR?

**Comment by nshoes** [comment]
@elinol bump on @joshk's question! :) Happy to review if this is good to go.

**Comment by joshk** [comment]
@elinol oh yeah, I noticed that this is still in Draft, is this ready for review?

**Review by joshk — COMMENTED** [review]
Do we need any tests for this?

**Comment by elinol** [comment]
oh, this has totally been under my radar! Don't think it's ready, will go through it now and mark as ready when done. Thanks @joshk and @nshoes for reminding me

**Comment by elinol** [comment]
@joshk I don't remember why there's no filters selectors added to the UI, did we wanna wait for the new UI? I could add them before merging this, or after if preferred.

**Comment by joshk** [comment]
I think it might be good to add it to the current UI, as the new UI is still a little bit away.

File: lib/nerves_hub/devices.ex [hunk 1]
```diff
@@ -221,10 +225,38 @@ defmodule NervesHub.Devices do
           else
             query
           end
+
+        {_, _} ->
```
[the remark this answers is not in the document]
Comment (elinol) [reply lib/nerves_hub/devices.ex:252]: We will have to handle the cases when not all metrics data needed is provided, like when only the key is selected. So either add 
```
{:metrics_operator, _} ->
    query

{:metrics_key, _} ->
    query
```
or do some kind of manipulation of filters in the liveview, but that might be a little bit too involved. Or, if you have any, even better, ideas I'm all ears.

[the remark this answers is not in the document]
Comment (elinol) [reply lib/nerves_hub/devices.ex:252]: @joshk what do you think?

**Review by joshk — APPROVED** [review]
Greeeeeeat work!

**Pull request merged**